src/soep_preparation/create_derived_variables/task.py

```diff
@@ -0,0 +1,174 @@
+"""Functions to create derived variables."""
+
+from pathlib import Path
+from typing import Annotated, Any
+
+import pandas as pd
+from pytask import PickleNode, task
+
+from soep_preparation.config import DATA_CATALOGS, SRC
+from soep_preparation.utilities.error_handling import fail_if_invalid_input
+from soep_preparation.utilities.general import (
+    get_file_names,
+    get_script_names,
+    load_module,
+)
+
+
+def _fail_if_too_many_or_few_dataframes(dataframes: dict, expected_entries: int):
+    if len(dataframes.keys()) != expected_entries:
+        msg = f"Expected {expected_entries} dataframes, got {len(dataframes.keys())}"
+        raise ValueError(
+            msg,
+        )
+
+
+def _get_relevant_data_files_mapping(
+    function_: Any,
+) -> dict[str, PickleNode]:
+    # arguments to the function (data files required for the variable)
+    data_names = [
+        data_name
+        for data_name in function_.__annotations__
+        if data_name in DATA_CATALOGS["data_files"]
+    ]
+    # return a mapping of the data file names to the corresponding dataframes
+    return {
+        data_name: DATA_CATALOGS["derived_variables"][data_name]
+        for data_name in data_names
+    }
+
+
+def _get_variable_names_in_module(module: Any) -> list[str]:
+    """Get the variable names in the module.
+
+    Args:
+        module: The module to get the variable names from.
+
+    Returns:
+        The variable names in the module.
+    """
+    return [
+        variable_name.split("derive_")[-1]
+        for variable_name in module.__dict__
+        if variable_name.startswith("derive_")
+    ]
+
+
+file_names = get_file_names(directory=SRC / "create_derived_variables")
```

Comment:
Not sure what are files, what are scripts? But let's postpone that one for sure. Just make a note that the logic here needs clarification so we won't forget.